src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -897,27 +882,20 @@ public SocketError ReceiveFrom(byte[] buffer, int offset, int count, ref SocketF
             }
         }
 
-        public SocketError ReceiveFromAsync(byte[] buffer, int offset, int count, SocketFlags flags, byte[] socketAddress, int socketAddressLen, Action<int, byte[], int, SocketFlags, SocketError> callback)
+        public SocketError ReceiveFromAsync(byte[] buffer, int offset, int count, SocketFlags flags, byte[] socketAddress, ref int socketAddressLen, out int bytesReceived, out SocketFlags receivedFlags, Action<int, byte[], int, SocketFlags, SocketError> callback)
         {
             SetNonBlocking();
 
             lock (_receiveLock)
             {
-                int bytesReceived;
-                SocketFlags receivedFlags;
+                bytesReceived = 0;
+                receivedFlags = SocketFlags.None;
```

Comment:
Should this initialization now be moved down to after the subsequent if?  We don't need to set these prior to that, right?